File: weeks/week_2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// FUNCTION PROTOTYPES
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

// MAIN
int main(void)
{
    string prompt = get_string("Text: ");

    int letters = count_letters(prompt);
    int words = count_words(prompt);
    int sentences = count_sentences(prompt);

    //test printf
    //printf("%i letters\n", letters);
    //printf("%i words\n", words);
    // printf("%i sentences\n", sentences);

    float L = ((float)letters / words) * 100;
    float S = ((float)sentences / words) * 100;

    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index < 16 && index >= 1)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }

}

int count_letters(string text)
{
    int total = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            total++;
        }
    }
    return total;
}
// count words
int count_words(string text)
{
    int total = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            total++;
        }
    }
    return total;
}
// count sentences
int count_sentences(string text)
{
    int total = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] == '.' || text[i] == '!' || text[i] == '?') && (text[i + 1] == ' '))
        {
            total++;
        }
    }
    return total;
}

File: weeks/week_6/sentimental-readability/readability.py
from cs50 import get_string
import math


# functions prototypes :

# Count letters
def nb_letters(text):
    total = 0
    for char in text:
        if char.isalpha():
            total += 1
    return total

# count words


def nb_words(text):
    total = len(text.split())
    return total


# count sentences
def nb_sentences(text):
    total = 0
    for i in range(len(text)):
        if (text[i] == '.' or text[i] == '!' or text[i] == '?') and (i == len(text) - 1 or text[i + 1] == ' '):
            total += 1
    return total


def main():

    # prompt
    prompt = get_string("Prompt: ")

    # call functions
    letters = nb_letters(prompt)
    words = nb_words(prompt)
    sentences = nb_sentences(prompt)

    # calculate index 1/2
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # calculate index 2/2
    index = round((0.0588 * L) - (0.296 * S) - 15.8)

    # print index depending on score
    if index < 1:
        print("Before Grade 1")
    elif index < 16:
        print(f"Grade {index}")
    else:
        print("Grade 16+")


main()